internal/postprocessor/main.go

```diff
@@ -268,3 +246,177 @@ func docURL(cloudDir, importPath string) (string, error) {
+			return "", "", err
+		}
+
+		if newPRTitle == "" {
+			newPRTitle = updateCommitTitle(title, commitPkg)
+			continue
+		}
+		newCommitTitle := updateCommitTitle(commitTitle, commitPkg)
+		bodySlice[commitTitleIndex] = newCommitTitle
+	}
+	body = strings.Join(bodySlice, "\n")
+	return newPRTitle, body, nil
+}
+
+func (c *config) getPR(ctx context.Context) (*github.PullRequest, error) {
+	client := github.NewClient(nil)
+	prs, _, err := client.PullRequests.List(ctx, c.githubUsername, "google-cloud-go", nil)
+	if err != nil {
+		return nil, err
+	}
+	pr, err := c.findValidPR(ctx, prs)
+	if err != nil {
+		return nil, err
+	}
+	return pr, nil
+}
+
+func (c *config) findValidPR(ctx context.Context, prs []*github.PullRequest) (*github.PullRequest, error) {
+	for _, pr := range prs {
+		if strings.Contains(*pr.Head.Label, c.branchPrefix) {
+			owlbotPR := pr
+			return owlbotPR, nil
+		}
+	}
+	return nil, errors.New("no PR found")
+}
+
+func (c *config) getScope(line string) (string, error) {
+	hash := extractHashFromLine(line)
+	commitPkg, err := c.getScopeFromGoogleapisCommitHash(hash)
+	if err != nil {
+		return "", err
+	}
+	return commitPkg, nil
+}
+
+func (c *config) getScopeFromGoogleapisCommitHash(commitHash string) (string, error) {
+	files, err := c.filesChanged(commitHash)
+	if err != nil {
+		return "", err
+	}
+	// if no files changed, return empty string
+	if len(files) == 0 {
+		return "", nil
+	}
+	scopesMap := make(map[string]bool)
+	scopes := []string{}
+	for _, filePath := range files {
+		for _, config := range generator.MicrogenGapicConfigs {
+			if config.InputDirectoryPath == filepath.Dir(filePath) {
+				scope := config.Pkg
+				if _, value := scopesMap[scope]; !value {
+					scopesMap[scope] = true
+					scopes = append(scopes, scope)
+				}
+				break
+			}
+		}
+	}
+	// if no in-scope packages are found or if many packages found, return empty string
+	if len(scopes) != 1 {
+		return "", nil
+	}
+	// if sincle scope found, return
+	return scopes[0], nil
+}
+
+// filesChanged returns a list of files changed in a commit for the provdied
+// hash in the given gitDir. Copied fromm google-cloud-go/gapicgen/git/git.go
+func (c *config) filesChanged(hash string) ([]string, error) {
+	out := execv.Command("git", "show", "--pretty=format:", "--name-only", hash)
+	out.Dir = c.googleapisDir
+	b, err := out.Output()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(string(b), "\n"), nil
+}
+
+func extractHashFromLine(line string) string {
+	hash := fmt.Sprintf("${%s}", hashFromLinePattern.SubexpNames()[1])
+	hashVal := hashFromLinePattern.ReplaceAllString(line, hash)
+	
+	return hashVal
+}
+
+func updateCommitTitle(title, titlePkg string) string {
+	var newTitle string
+
+	titleFirstPart := fmt.Sprintf("${%s}", firstPartTitlePattern.SubexpNames()[1])
+	titleSecondPart := fmt.Sprintf("${%s}", secondPartTitlePattern.SubexpNames()[1])
+
+	firstTitlePart := firstPartTitlePattern.ReplaceAllString(title, titleFirstPart)
```

Comment:
nit: can we change the variable names here to be more clear. Having both firstTitlePart and titleFirstPart is tricky to read